src/generated.rs

```diff
@@ -0,0 +1,59 @@
+    map.insert(vec![2049], vec![0,0,0,0,0,0,0,0,0,0,0,2,0,1,134,163,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]);
+    map.insert(vec![2123,2152], vec![50,1,0,4,0,0,66,0,19,55,0,0]);
+    map.insert(vec![2302], vec![0,2,241,38,1,38,240,144,166,240,38,87,78,172,160,236,248,104,228,141,33]);
+    map.insert(vec![3283], vec![1,64,1,3]);
+    map.insert(vec![3386], vec![78,1,0,4,222,254,200,0]);
+    map.insert(vec![3478], vec![0,1,0,0,33,18,164,66,0,0,0,0,0,0,0,0,0,0,0,0]);
+    map.insert(vec![3784], vec![1,231,229,117,49,163,23,11,33,207,191,43,153,78,221,25,172,222,8,95,139,36,10,17,25,182,115,111,173,40,19,210,10,185,18,117]);
+    map.insert(vec![4665,4666,4672,6429], vec![228,96]);
+    map.insert(vec![5351], vec![0,0]);
+    map.insert(vec![5353], vec![0,0,0,0,0,1,0,0,0,0,0,0,9,236,224,125,208,77,5,202,0,0,12,0,1]);
+    map.insert(vec![5632], vec![]);
+    map.insert(vec![5683], vec![1,1,206,187,224,76]);
+    map.insert(vec![6481], vec![190,0,0]);
+    map.insert(vec![7777], vec![224]);
+    map.insert(vec![8767], vec![244,190,3,0,0,0,0,0,0,0,0,0,1,0,0,0,43,168,94,174,160,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,13,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,8,202,224,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]);
+    map.insert(vec![9987], vec![5,202,127,22,156,17,249,137,0,0,0,0,2,157,116,139,69,170,123,239,185,158,254,173,8,25,186,207,65,224,22,162,50,108,243,207,244,142,60,68,131,200,141,81,69,111,144,149,35,62,0,151,43,28,113,178,78,192,97,241,215,111,197,126,246,72,82,191,130,106,162,59,101,170,24,122,23,56,195,129,39,195,71,252,167,53,186,252,15,157,157,114,36,157,252,2,23,109,107,177,45,114,198,227,23,28,149,217,105,153,87,206,221,223,5,220,3,148,86,4,58,20,229,173,154,43,20,48,58,35,163,37,173,232,230,57,138,133,42,198,223,229,93,45,160,47,93,156,215,43,36,251,176,156,194,186,137,180,27,23,162,182]);
+    map.insert(vec![10001], vec![1,0,0,0]);
+    map.insert(vec![10080], vec![170,218,38,234,222,0,0,0,0,0,14,14,206]);
+    map.insert(vec![11211], vec![0,16,0,1,0]);
+    map.insert(vec![17185], vec![114,55,114,55,0,0,0,0,0,0,0,2,85,85,85,85,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,255,255,85,16,0,0,0,60,0,0,0,3,0,0,0,2,0,0,0,0,0,0,0,0]);
+    map.insert(vec![26000,26001,26002,26003,27960,27961,27962,27963,30720,30721,30722,30723,44400], vec![255,255,255,255,234]);
+    map.insert(vec![27015,27016,27017,27018,27019,27020,27021,27022,27023,27024,27025,27026,27027,27028,27029], vec![255,255,255,255,206,238,224]);
+    map.insert(vec![27444], vec![68,173]);
+    map.insert(vec![27910,27911,27912,27913], vec![255,255,255,255]);
+    map.insert(vec![31337], vec![206,99,209,210,22,231,19,207,56,165,165,134,178,117,75,153,170,50,88]);
+    map.insert(vec![34555], vec![]);
+    map.insert(vec![64738], vec![0,0,0,0,171,205,239]);
+    map
```

Comment:
this is a big file, do we have to make this at run time? can we not build this before hand and consume it with yaml or something? or even have a perfect hash map https://skerritt.blog/perfect-hash-tables/ ?

Reply:
> I am not a super big fan of using Rust to build files :( We should use something like Bash or Python I think, as it's easier to read & contribute to?

Do you mean rewriting the entire file parser in Python?

I not entirely sure how this could be done, maybe you know @HDTHREE

> this is a big file, do we have to make this at run time? can we not build this before hand and consume it with yaml or something? or even have a perfect hash map https://skerritt.blog/perfect-hash-tables/ ?

A couple of things could happen with this file; we could .gitignore it. I cannot get any version of the Rust LSP to format it for me, so I think it breaks part of the CI. I was super weirded out by the whole code generation at compile time thing, but yes the way we are actually consuming this map at runtime is something I can work on. 

> @BrendanGlancy, I'm away for a few days but will take a look ASAP. Looks like the build is failing at the moment though so probably worth determining where that's coming from.

@PsypherPunk && Bee no rush on any of this, just wanted to get your thoughts on this before I moved any further forward as this PR is already the size of the Epstein list.

Reply:
> Do you mean rewriting the entire file parser in Python?

I am a bit sick so I didn't take a proper look, but I think your `build.rs` parses the nmap file and turns it into another Rust file to use, right?

typically for scripts like that I see them as Python / Bash and they go into a seperate `/scripts` folder or something or maybe `fixtures`  🤔 

> A couple of things could happen with this file; we could .gitignore it. I cannot get any version of the Rust LSP to format it for me, so I think it breaks part of the CI. I was super weirded out by the whole code generation at compile time thing, but yes the way we are actually consuming and creating this map at runtime is something I can work on.

yeah that weirds me out too, i think generally the PR looks fine though :)

> @PsypherPunk && Bee no rush on any of this, just wanted to get your thoughts on this before I moved any further forward as this PR is already the size of the Epstein list.

PR is failing partially because cargo format cant format `build,rs`, another win for not putting it into rust ;)

i think trying one of these:
https://www.reddit.com/r/rust/comments/1c6ceuu/how_to_force_cargo_fmt_to_ignore_some_files/

and then running cargo format locally to see if it works or not.

other issue is casued by https://github.com/RustScan/RustScan/issues/527

this has a fix:

The temporary solution is
change
let arguments = shell_words::split(&to_run).expect("Failed to parse script arguments");
to
let arguments:Vec<String> =to_run.split_whitespace().map(|s| s.to_string()).collect();

does that work? 🤔

Reply:
> I am a bit sick so I didn't take a proper look, but I think your build.rs parses the nmap file and turns it into another Rust file to use, right?

> typically for scripts like that I see them as Python / Bash and they go into a seperate /scripts folder or something or maybe fixtures 🤔

The problem with using python for this is, and I am not 100% sure about this, we will not be able to put the values for the BTreeMap (or whatever datastructure) during compile time.